test/e2e/cassandracluster_test.go

```diff
@@ -144,3 +152,179 @@ func cassandraClusterRollingRestartDCTest(t *testing.T, f *framework.Framework,
 
 	assert.Equal(t, false, cc.Spec.Topology.DC[0].Rack[0].RollingRestart)
 }
+
+func cassandraClusterServiceTest(t *testing.T, f *framework.Framework, ctx *framework.TestCtx) {
+	namespace, err := ctx.GetNamespace()
+	if err != nil{
+		t.Fatalf("Could not get namespace: %v", err)
+	}
+
+	kind := "CassandraCluster"
+	apiVersion := "db.orange.com/v1alpha1"
+
+	cluster := &api.CassandraCluster{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       kind,
+			APIVersion: apiVersion,
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "service-test",
+			Namespace: namespace,
+			Labels:    map[string]string{"cluster": "k8s.pic"},
+		},
+		Spec: api.CassandraClusterSpec{
+			BaseImage: "orangeopensource/cassandra-image",
+			NodesPerRacks: 1,
+			DeletePVC: true,
+			Resources: api.CassandraResources{
+				Limits: api.CPUAndMem{
+					CPU: "500m",
+					Memory: "1Gi",
+				},
+			},
+			Topology: api.Topology{
+				DC: api.DCSlice{
+					api.DC{
+						Name: "dc1",
+						Rack: api.RackSlice{
+							api.Rack{
+								Name: "rack1",
+							},
+						},
+					},
+					api.DC{
+						Name: "dc2",
+						Rack: api.RackSlice{
+							api.Rack{
+								Name: "rack1",
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	log.Debugf("Creating cluster")
+	if err := f.Client.Create(goctx.TODO(), cluster, NoCleanup()); err != nil && !apierrors.IsAlreadyExists(err) {
+		t.Fatalf("Error Creating CassandraCluster: %v", err)
+	}
+
+	waitForClusterToBeReady(cluster, f, t)
+
+	cluster = getCassandraCluster("service-test", "cassandra-e2e", f, t)
+
+	services, err := listServices(namespace, metav1.ListOptions{
+		LabelSelector: labels.FormatLabels(map[string]string{
+			"app": "cassandracluster",
+			"cassandracluster": cluster.Name,
+		}),
+	}, f)
+	if err != nil {
+		t.Errorf("Error listing services: %v", err)
+	}
+	assert.Equal(t, 2, len(services.Items))
+
+	var clusterService, monitoringService v1.Service
+	if val, ok := services.Items[0].Labels["k8s-app"]; ok {
+		assert.Equal(t, "exporter-cassandra-jmx", val)
+		clusterService = services.Items[0]
+		monitoringService = services.Items[1]
+	} else if val, ok := services.Items[1].Labels["k8s-app"]; ok {
+		assert.Equal(t, "exporter-cassandra-jmx", val)
+		clusterService = services.Items[0]
+		monitoringService = services.Items[1]
+	} else {
+		t.Errorf("Failed to find monitoring service. Found: %v", services.Items[0])
+	}
+
+	assert.Equal(t, 1, len(clusterService.ObjectMeta.OwnerReferences))
+	assert.Equal(t, kind, clusterService.ObjectMeta.OwnerReferences[0].Kind)
+	assert.Equal(t, cluster.Name, clusterService.ObjectMeta.OwnerReferences[0].Name)
+	assert.True(t, *clusterService.ObjectMeta.OwnerReferences[0].Controller)
+	assert.Equal(t, 3, len(clusterService.Spec.Ports))
+
+	assertServiceExposesPort(t, &clusterService, "cassandra-port", 9042)
+	assertServiceExposesPort(t, &clusterService, "cassandra-thrift", 9160)
```

Comment:
I still think we should stop supporting thrift. what do you think @jal06, @allamand ?

Reply:
I have remove thrift from CassKop in a separate PR. we should also remove it from tests